src/sock.c

```diff
@@ -602,11 +604,11 @@ static void mg_iotest(struct mg_mgr *mgr, int ms) {
     if (c->is_closing) tvp = &tv_zero;
   }
 
-  if ((rc = select((int) maxfd + 1, &rset, &wset, &eset, tvp)) < 0) {
+  if ((rc = select((int) maxfd + 1, &rset, &wset, &eset, tvp)) <= 0) {
```

Comment:
This affects all other systems, for example, rc = 0 means that Windows will now check maxfd (it did not before), and all will execure FD_ZERO (they did not before).
I would introduce a special case for 0 with MG_ARCH == MG_ARCH_THREADX. wdyt ?